### PR TITLE
greenspace + refactor

### PR DESCRIPTION
**Pull request by nordbergmikael**

### CODE REVIEW

File: internal/pkg/application/transform/transformers.go [hunk 1]
```diff
@@ -132,10 +152,88 @@ func AirQualityObserved(ctx context.Context, msg iotcore.MessageAccepted, cbClie
...
-	v, ok := msg.GetBool(measurements.Presence)
+func IndoorEnvironmentObserved(ctx context.Context, msg core.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	properties := []entities.EntityDecoratorFunc{
+		entities.DefaultContext(),
+		Location(msg.Latitude(), msg.Longitude()),
+		DateObserved(msg.Timestamp),
+	}
+
+	/*
+		ObjectURN: urn:oma:lwm2m:ext:3303
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers.go]: if illuminance exists as its own type, perhaps we ought to add it in here with this comment for the temp and airquality

File: internal/pkg/application/transform/transformers.go [hunk 2]
```diff
@@ -241,71 +353,117 @@ func WaterConsumptionObserved(ctx context.Context, msg iotcore.MessageAccepted,
...
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
+	buildfragment := func(patchProperties ...entities.EntityDecoratorFunc) error {
+		fragment, err := entities.NewFragment(patchProperties...)
+		if err != nil {
+			return fmt.Errorf("entities.NewFragment failed: %w", err)
+		}
+
+		properties := append(patchProperties, DateObserved(curDateTime))
+
+		//_, err = cbClient.UpdateEntityAttributes(ctx, entityID, fragment, headers)
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers.go]: this should be removed, no?

File: internal/pkg/application/transform/transformers.go [hunk 3]
```diff
@@ -241,71 +353,117 @@ func WaterConsumptionObserved(ctx context.Context, msg iotcore.MessageAccepted,
...
+		_, err = cbClient.MergeEntity(ctx, entityID, fragment, headers)
+
+		if err != nil {
+			// If we failed to update the entity's attributes, we need to create it
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers.go:409]: check here if the error is that we can't find the entity, as we do it in IndoorEnvironmentObserved above. we've used that same pattern in another service as well. Line 202 onwards.

File: internal/pkg/application/transform/transformers.go [hunk 4]
```diff
@@ -241,71 +353,117 @@ func WaterConsumptionObserved(ctx context.Context, msg iotcore.MessageAccepted,
...
 		}
-	} else {
-		log.Debug().Msgf("no volume property was found in message from %s", msg.Sensor)
+
+		return err
 	}
 
-	return nil
-}
+	// GreenspaceRecord is called by one of its properties. First out creates the entity, all other subsequent calls, independent which property, updates the entity.
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers.go]: I'm a little unsure of what this comment means, but I think it should probably say "GreenspaceRecord is called by one of its properties. First it creates the entity, and on all subsequent calls, independent of which property, it updates the entity."

File: internal/pkg/application/transform/transformers_test.go [hunk 5]
```diff
@@ -196,33 +217,73 @@ func TestThatWaterConsumptionObservedIsCreatedIfNonExisting(t *testing.T) {
 	is.Equal(string(b), expectedCreateBody)
 }
 
-func TestDeltaVolumes(t *testing.T) {
-	is := is.New(t)
+// GreenspaceRecord test notes:
+// Pressure and Condctivity may come as array of values from iot-core.
+//  - first occurances of these are treated as primary measurement
+//	- subsequent occuranceses are treated as additional measurements and ignored
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers_test.go]: sp. of occurance on line 222 - should be singular. "first occurance is treated.."

on line 223, should be occurances.

File: internal/pkg/application/transform/transformers_test.go [hunk 6]
```diff
@@ -196,33 +217,73 @@ func TestThatWaterConsumptionObservedIsCreatedIfNonExisting(t *testing.T) {
...
 
-	v := 100.009
-	dv1 := 10.0
+func TestThatGreenspaceRecordIsCreatedIfNonExisting(t *testing.T) {
```
Comment (shestyf) [inline internal/pkg/application/transform/transformers_test.go]: NonExistant

**Review by shestyf — CHANGES_REQUESTED** [review]
spelling errors mostly

**Review by istyf — COMMENTED** [review]
Nothing to add.

**Review by istyf — APPROVED** [review]
/lgtm

**Review by shestyf — APPROVED** [review]
/lgtm

**Pull request merged**